packages/cli/src/main/kotlin/elide/tool/cli/cmd/repl/ToolShellCommand.kt

```diff
@@ -275,27 +286,19 @@ import org.graalvm.polyglot.Engine as VMEngine
     )
     internal var sources: List<String> = emptyList()
 
-    /** Apply configuration to the VM based on the provided arguments. */
-    internal fun apply(debug: Boolean): Stream<VMProperty> {
-      return (if (!debug) emptyList<VMProperty>() else listOfNotNull(
-        // if specified, add custom `path`
-        when (val p = path) {
-          null -> null
-          else -> VMStaticProperty.of("inspect.Path", p)
-        },
-      ).plus(sources.joinToString(",").let { targets ->
-        // format + add any custom source directories
-        if (targets.isEmpty()) emptyList() else listOf(
-          VMStaticProperty.of("inspect.SourcePath", targets)
-        )
-      })).stream()
+    /** Apply these settings to the root engine configuration container. */
+    internal fun apply(config: PolyglotEngineConfiguration) {
```

Comment:
Instead of returning a property stream, this method now directly configures the root of the DSL. It is used in the `configureEngine` event.